src/fork.rs

```diff
@@ -240,10 +249,18 @@ impl ForkDetails<HttpForkSource> {
         let l1_batch_number = block_details.l1_batch_number;
 
         println!(
-            "Creating fork from {:?} L1 block: {:?} L2 block: {:?} with timestamp {:?} and L1 gas price {:?}",
-            url, l1_batch_number, miniblock, block_details.base.timestamp, block_details.base.l1_gas_price,
+            "Creating fork from {:?} L1 block: {:?} L2 block: {:?} with timestamp {:?}, L1 gas price {:?} and protocol version: {:?}" ,
+            url, l1_batch_number, miniblock, block_details.base.timestamp, block_details.base.l1_gas_price, block_details.protocol_version
         );
 
+        if !block_details
+            .protocol_version
+            .map(supported_protocol_versions)
+            .unwrap_or(false)
+        {
+            panic!("This block is using the unsupported protocol version: {:?}. This binary supports versions 13-15.", block_details.protocol_version);
```

Comment:
Should we move supported protocol versions to a list so we don't have to hard-code `13-15` here? Example code curtesy of ChatGPT:

```rust
// Step 1: Define a list of supported versions
const SUPPORTED_VERSIONS: &[ProtocolVersionId] = &[
    ProtocolVersionId::Version13,
    ProtocolVersionId::Version14,
    ProtocolVersionId::Version15,
];

// Step 2: Modify the function
pub fn supported_protocol_versions(version: ProtocolVersionId) -> bool {
    SUPPORTED_VERSIONS.contains(&version)
}

// Step 3: Utility function
pub fn supported_versions_to_string() -> String {
    let versions: Vec<String> = SUPPORTED_VERSIONS.iter()
        .map(|v| format!("{:?}", v))
        .collect();
    versions.join(", ")
}

// Step 4: Use in panic
panic!(
    "This block is using the unsupported protocol version: {:?}. This binary supports versions {}.",
    block_details.protocol_version,
    supported_versions_to_string()
);

``

Reply:
nice. idea. Done.